src/item.h

```diff
@@ -1222,13 +1222,20 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
         bool gunmod_compatible( const item& mod, bool alert = true, bool effects = true ) const;
 
         struct gun_mode {
-            std::string mode; /** name of this mode */
-            item *target;     /** pointer to base gun or attached gunmod */
-            int qty;          /** burst size or melee range */
+            std::string mode;           /** name of this mode */
+            item *target = nullptr;     /** pointer to base gun or attached gunmod */
+            int qty = 0;                /** burst size or melee range */
 
             /** flags change behavior of gun mode and are **not** equivalent to item flags */
             std::set<std::string> flags;
 
+            gun_mode() = default;
+            gun_mode( const std::string &mode, item *target, int qty, const std::set<std::string> &flags ) :
+                mode( mode ),
```

Comment:
Initializer lists (obviously) initialize member variables, which are defined one per line. The list items and the variables should match (both order and usually number too). If you want to make sure (visually) they're in the right order, or reorder them, that's simply easier when they formatted the same way. That's enough for me to state the former and the latter.